docker/image_types.yaml

```diff
@@ -17,54 +17,116 @@
   depends_on:
     - 'base-image'
 
-'coverage-builder':
-  tag: 'builders/coverage'
-  context: 'fuzzers/coverage'
-  dockerfile: 'fuzzers/coverage/builder.Dockerfile'
+'dispatcher-image':
+  tag: 'dispatcher-image'
+  context: 'docker/dispatcher-image'
+  depends_on:
+    - 'base-image'
+
+# '.coverage-builder':
+#   tag: 'builders/coverage'
+#   context: 'fuzzers/coverage'
+#   dockerfile: 'fuzzers/coverage/builder.Dockerfile'
+#   depends_on:
+#     - 'base-builder'
+
+# '.coverage-{benchmark}-builder':
+#   tag: 'builders/coverage/{benchmark}'
+#   context: '.'
+#   dockerfile: 'docker/benchmark-builder/Dockerfile'
+#   build_arg:
+#     - 'fuzzer=coverage'
+#     - 'benchmark={benchmark}'
+#   depends_on:
+#     - 'coverage-builder'
+
+'.{fuzzer}-builder':
+  tag: 'builders/{fuzzer}'
+  context: 'fuzzers/{fuzzer}'
+  dockerfile: 'fuzzers/{fuzzer}/builder.Dockerfile'
   depends_on:
     - 'base-builder'
 
-'coverage-{benchmark}-builder':
-  tag: 'builders/coverage/{benchmark}'
+'.{fuzzer}-{benchmark}-builder':
+  tag: 'builders/{fuzzer}/{benchmark}'
   context: '.'
   dockerfile: 'docker/benchmark-builder/Dockerfile'
   build_arg:
-    - 'fuzzer=coverage'
+    - 'fuzzer={fuzzer}'
     - 'benchmark={benchmark}'
   depends_on:
-    - 'coverage-builder'
+    - '.{fuzzer}-builder'
 
-'{fuzzer}-builder':
-  tag: 'builders/{fuzzer}'
+'.{fuzzer}-{benchmark}-intermediate-runner':
+  tag: 'runners/{fuzzer}/{benchmark}-intermediate'
+  context: 'fuzzers/{fuzzer}'
+  dockerfile: 'fuzzers/{fuzzer}/runner.Dockerfile'
+  depends_on:
+    - 'base-runner'
+
+'.{fuzzer}-{benchmark}-runner':
```

Comment:
will this be instantiated with fuzzer=coverage?

Reply:
For the coverage images, yes.

Reply:
That's a problem, right? We don't have "runner" versions of coverage images, only "build" versions. This is what we can avoid with the coverage- image types (that this pr currently comments out).

Reply:
No, it skips the coverage runners in the generation logic using a conditional.